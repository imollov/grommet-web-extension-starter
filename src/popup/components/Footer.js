import React from 'react';
import browser from 'webextension-polyfill';
import { Anchor, Footer } from 'grommet';

const handleClick = e => {
  e.preventDefault();
  browser.tabs.create({ active: true, url: e.target.href });
};

export default () => (
  <Footer background="brand" pad="medium">
    <Anchor
      label="Docs"
      href="https://v2.grommet.io/docs"
      onClick={handleClick}
    />
    <Anchor
      label="Components"
      href="https://v2.grommet.io/components"
      onClick={handleClick}
    />
    <Anchor
      label="Storybook"
      href="https://storybook.grommet.io"
      onClick={handleClick}
    />
  </Footer>
);
