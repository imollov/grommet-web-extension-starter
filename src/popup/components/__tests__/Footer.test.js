import React from 'react';
import { shallow, mount } from 'enzyme';

import Footer from '../Footer';

describe('Footer Component', () => {
  it('smoke test', () => {
    const wrapper = shallow(<Footer />);

    expect(wrapper).toExist();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Footer />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call webextension API', () => {
    const wrapper = mount(<Footer />);

    const link = wrapper.findWhere(
      el => el.text() === 'Docs' && el.type() === 'a',
    );
    link.simulate('click');

    expect(global.browser.tabs.create).toHaveBeenCalled();
  });
});
