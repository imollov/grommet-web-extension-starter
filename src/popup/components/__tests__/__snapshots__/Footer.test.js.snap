// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match snapshot 1`] = `
<Footer
  background="brand"
  pad="medium"
>
  <Anchor
    href="https://v2.grommet.io/docs"
    label="Docs"
    onClick={[Function]}
  />
  <Anchor
    href="https://v2.grommet.io/components"
    label="Components"
    onClick={[Function]}
  />
  <Anchor
    href="https://storybook.grommet.io"
    label="Storybook"
    onClick={[Function]}
  />
</Footer>
`;
